Changes

```diff
@@ -124,6 +124,11 @@ Working version
   (Samuel Thibault, review by Antonin Décimo, Sébastien Hinderer and Miod
    Vallat)
 
+- #12628: Improved error message for unsafe values print out the full path for
```

Comment:
A comma seems to be missing
> Improved error message for unsafe values, print out the full path for

It is also generally better to start with a more generic description and not expect the reader to know or remember what are `unsafe values`. We 

> #12628: Improved error message for recursive modules with an "unsafe" cycle ,  print out the full path for unsafe items

We could also avoid  with explaining the mechanics of the error message ?

> #12628: Improved error message for recursive modules with an "unsafe" cycle  when the unsafe items
   appear inside submodules.
   
 Please feel free to pick the description that you like the best.